src/main/java/org/opensearch/security/securityconf/DynamicConfigModelV7.java

```diff
@@ -312,6 +314,12 @@ private void buildAAA() {
             }
         }
 
+        Settings oboSettings = getDynamicOnBehalfOfSettings();
+        if (oboSettings.get("signing_key") != null && oboSettings.get("encryption_key") != null) {
+            final AuthDomain _ad = new AuthDomain(new NoOpAuthenticationBackend(Settings.EMPTY, null), new OnBehalfOfAuthenticator(getDynamicOnBehalfOfSettings()), false, -1);
```

Comment:
⚠️ NoOpAuthenticationBackend always returns a user object even if they don't exist - that's a problem

Reply:
On behalf of tokens are only supported for internal users?